auslib/blobs/apprelease.py

```diff
@@ -929,6 +936,162 @@ def getReferencedReleases(self):
+                # If there are no overlapping fields between the groups, there can't be a conflict
+                field_dupes = set(group1["fields"].keys()).intersection(group2["fields"].keys())
+                if len(field_dupes) == 0:
+                    continue
+                # If there are overlapping fields, we need to see if what they apply to overlap in any way
+                condition_results = []
+                for cond in ("channels", "locales", "versions"):
+                    matches = False
+                    # If a condition is not specified for a group, it always matches anything
+                    if cond not in group1["for"] or cond not in group2["for"]:
+                        matches = True
+                    else:
+                        if cond == "channels":
+                            for (value1, value2) in itertools.product(group1["for"][cond], group2["for"][cond]):
+                                # Exact match of concrete channel or two globs
+                                if value1 == value2:
+                                    matches = True
+                                    break
+                                # Logical match in either direction, which takes into account
+                                # globbing and fallback channels
+                                elif matchChannel(value1, value2.rstrip("*"), getFallbackChannel(value2.rstrip("*"))):
+                                    matches = True
+                                    break
+                                elif matchChannel(value2, value1.rstrip("*"), getFallbackChannel(value1.rstrip("*"))):
+                                    matches = True
+                                    break
+                        elif cond == "locales":
+                            if set(group1["for"][cond]).intersection(set(group2["for"][cond])):
+                                matches = True
+                        elif cond == "versions":
+                            for (value1, value2) in itertools.product(group1["for"][cond], group2["for"][cond]):
+                                # Check for exact matches - any exact match between two concrete
+                                # versions or two comparisons is a match.
+                                if value1 == value2:
+                                    matches = True
+                                    break
+                                # Also check for matches with version comparison involved
+                                elif matchVersion(value1, strip_operator(value2)):
+                                    matches = True
+                                    break
+                                elif matchVersion(value2, strip_operator(value1)):
+                                    matches = True
+                                    break
+
+                    condition_results.append(matches)
+
+                if all(condition_results):
```

Comment:
Feels like all() should be any(), but I didn't get to the tests yet.

Reply:
I think you're right, and it looks like a gap in the tests right now.

Reply:
Actually, `all` is correct because we have one entry in condition_results for each of (channels, versions, locales), and if any part of any of them matched, we get a True result into the list (and we need each part to be True to have an overall match).